core/src/main/java/bio/terra/pearl/core/service/participant/search/facets/sql/KeywordFacetSqlGenerator.java

```diff
@@ -0,0 +1,63 @@
+package bio.terra.pearl.core.service.participant.search.facets.sql;
+
+import bio.terra.pearl.core.dao.BaseJdbiDao;
+import bio.terra.pearl.core.service.participant.search.EnrolleeSearchUtils;
+import bio.terra.pearl.core.service.participant.search.facets.StringFacetValue;
+import org.jdbi.v3.core.statement.Query;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class KeywordFacetSqlGenerator implements FacetSqlGenerator<StringFacetValue> {
+
+    public KeywordFacetSqlGenerator() {}
+
+    @Override
+    public String getTableName() {
+        // this facet is only tied to enrollee and Profile tables that are already joined
+        return "";
+    }
+
+    @Override
+    public String getJoinQuery() {
+        return "";
+    }
+
+    @Override
+    public String getSelectQuery(StringFacetValue facetValue) {
+        return null; // already included in base query
+    }
+
+    @Override
+    public String getWhereClause(StringFacetValue facetValue, int facetIndex) {
+        if (facetValue.getValues().isEmpty()) {
+            return " 1 = 1";
+        }
+        return IntStream.range(0, facetValue.getValues().size())
+                .mapToObj(index -> {
+                    String paramName = EnrolleeSearchUtils.getSqlParamName("keyword", facetValue.getKeyName(), index);
+                    return """
+                            (profile.given_name ilike :%s 
+                              OR profile.family_name ilike :%s
+                              OR profile.contact_email ilike :%s
+                              OR enrollee.shortcode ilike :%s)
+                            """
+                            .formatted(paramName, paramName, paramName, paramName);
```

Comment:
I recently learned about referencing formatting arguments positionally which allows you to avoid repeating the value in cases like this:
```suggestion
                    return """
                            (profile.given_name ilike :%1$s 
                              OR profile.family_name ilike :%1$s
                              OR profile.contact_email ilike :%1$s
                              OR enrollee.shortcode ilike :%1$s)
                            """
                            .formatted(paramName);
```